js/gamestate.js

```diff
@@ -206,9 +183,23 @@ class GameState {
     if (player.mana < card.cost) {
       throw `need ${card.cost} mana but only have ${player.mana}`
     }
-    player.board.push(card)
-    player.hand.splice(from, 1)
-    player.mana -= card.cost
+    // it's a creature
+    if (card.defense) {
+      player.board.push(card)
+      player.hand.splice(from, 1)
+      player.mana -= card.cost      
+    } else if (card.beginTurn) { // it's a timewalk
+      player.trash.push(card)
+      player.hand.splice(from, 1)
+      player.mana -= card.cost      
+      for (let i=0;i<card.beginTurn;i++) {
+        let event = new CustomEvent("turnEnded", {});
+        window.dispatchEvent(event);
+        window.client.makeLocalMove({"op":"endTurn"});
```

Comment:
i dont think you want to access window.client from the state object. does this pass unit tests?